src/components/QueryEditor/QueryHeader.tsx

```diff
@@ -110,7 +111,7 @@ export const QueryHeader = (props: QueryEditorHeaderProps) => {
         }}
       />
       <FlexItem grow={1} />
-      <Button variant="primary" icon="play" size="sm" onClick={onRunQuery}>
+      <Button variant="primary" icon="play" size="sm" onClick={onRunQuery} aria-label="Run query">
```

Comment:
Are `aria-label`'s needed here given that these components have text labels already?

Reply:
yes, at least for the e2e tests to find them as it was in the legacy editor

Reply:
I thought this might be why 😊 It may be prudent to move these new labels to `data-testid` for consistency, wdyt?

Reply:
Previously it was using aria-labels but we can change it. It seems it doesn't work with the `InlineSelect` (it does not forward the `data-testid`) but it does for the `Button`.

Reply:
I've opened an issue [here](https://github.com/grafana/grafana-experimental/issues/54) for the `InlineSelect` component 😊